### PR TITLE
Manager enhancements and fixes

### PR DESCRIPTION
**Pull request by mikliapko**
Closes https://github.com/scylladb/scylla-manager/issues/4048

Addresses several things:
- Introduced the new test for restore benchmark (9-nodes cluster, 1TB 2 tables dataset);
- Done a refactoring for Manager jenkinsfiles (for removed jenkinsfiles their usage was checked - none of them is used in Manager related Jenkins jobs);
- Manager tests stabilization and fixes.

### Testing

- [x] [sanity stabilization](https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/mikita/job/manager-master/job/ubuntu22-vnodes-tablets-cluster-test/21/);
- [x] [restore benchmark](https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/mikita/job/manager-master/job/restore-benchmark-test-1tb-2tables-9nodes/1/) on 9-nodes cluster

### PR pre-checks (self review)

- [x] I added the relevant `backport` labels
- [x] I didn't leave commented-out/debugging code

### CODE REVIEW

File: defaults/manager_restore_benchmark_snapshots.yaml
Comment (soyacz) [inline defaults/manager_restore_benchmark_snapshots.yaml:11]: rc3 is available

**Review by soyacz — APPROVED** [review]

File: defaults/manager_restore_benchmark_snapshots.yaml
Comment (mikliapko) [reply defaults/manager_restore_benchmark_snapshots.yaml:11]: Yeah, but we are going to wait for official release and rebuild the snapshots.
The change here is typing fix as initially the version `...~rc0` was put there wrongly.

**Review by Michal-Leszczynski — APPROVED** [review]

**Comment by mikliapko** [comment]
@grzywin Could you please take a look?
I suppose we are still missing one review from QA team member.

**Pull request merged**
